### DISCUSSION TITLE
How to generate password in non-js non-ts way

### DISCUSSION THREAD

**Discussion by raidgar98** [jesec/flood#825] [Show and tell]
⚠️ I am not front-end developer

If you just want hash of new password, here is faster way of getting it: https://github.com/jesec/flood/discussions/825#discussioncomment-11341863

I spent like 1h to run  `argon2` and `argon2-cli`. Here is alternative in python

Here is script:

```python
from argon2 import PasswordHasher
print(PasswordHasher().hash("my_super_password"))
```

and that's it. Here is my python version: `3.10.12`

and here is setup:

```bash
python3 -m venv .venv
source .venv/bin/activate

# you can skip all the above if you don't care about where what is installed (⚠️ skipping installs it globally for user)
pip install argon2-cffi

# after putting this script to `script.py` run:
python script.py

# or just:
python 
# and paste it in console
```

Can't find online interpreter which allows sharing and installing packages for free, if you find one let me know and i will share it here for easy online password generation

**Comment by trim21** [comment]
why you need to generate password by yourself? 

This feels like a XY problem, what do you try to accomplish?

**Reply by raidgar98** [reply]
I had one generated by my password manager and accidentally override it. Had no way to update it

**Reply by trim21** [reply]
maybe you can backup current database and create a new user with known password, then copy the hash from new user ( I guess you know sqlite3). In my understanding then password of new user will become password of the old user.

**Reply by raidgar98** [reply]
also possible, didn't thought about. Just going with this FAQ: https://github.com/jesec/flood/wiki/F.A.Q.#i-forgot-my-password-how-can-i-recover-it

and later (referenced by FAQ): https://github.com/Flood-UI/flood/issues/529#issuecomment-640027600

**Reply by trim21** [reply]
I guess you can just create a new user and let floold generate hash for you.

**Reply by raidgar98** [reply]
i didn't want to create new user but you are right this is also good solution. I think i just blinded myself on this approach

**Reply by trim21** [reply]
by "create a new user" I mean to get hash, not mean to use new account.

**Reply by raidgar98** [reply]
I'll leave it for anyone who also will look for such problem, but i will edit it to be more pleasant and link to your comment in header as quickest solution

**Reply by raidgar98** [reply]
you mean to create a new user by `flood` binary inside docker?

**Reply by trim21** [reply]
sorry, I repeated myself, and looks like it cause some confusion. I still mean https://github.com/jesec/flood/discussions/825#discussioncomment-11341843

backup rundir and edit sqlite3 file inside it.

**Reply by raidgar98** [reply]
ok, but if I create new user in sqlite, does db will generate hash of new password? By creating new user you mean calling `INSERT INTO...` or i am missing something? sorry i don't understand how because from front-end side i have only:

![obraz](https://github.com/user-attachments/assets/811795af-39e5-4ce6-99f1-5fa0369a4dba)

**Reply by raidgar98** [reply]
oh i understand now, by backuping db (and tmp removing it from flood dir) i will go to configuration screen. Thanx

**Reply by trim21** [reply]
1. backup rundir, as A
2. mount a new path as docker rundir, as B
3. create a new user


now we have 2 sqlite3 database, A and B. hash of password in database B is known, use it to override old hash in database A.
